### PR TITLE
Send Perf Data to Explorer Frontend

### PR DESCRIPTION
**Pull request by vprajapati-tt**
- A few small changes to explorer to send perf data to the frontend.
- Changed `to_adapter_format` to allow for multiple objects
- Added `add_to_dataclass` to append perf_data to Graph dataclass.

### CODE REVIEW

File: tools/explorer/CMakeLists.txt [hunk 1]
```diff
@@ -18,7 +18,9 @@ ExternalProject_Add(
 add_custom_target(explorer
   COMMENT "Building tt-explorer... ${TTMLIR_BIN_DIR}"
   COMMAND pip install $<$<CONFIG:Debug>:-e> ${CMAKE_CURRENT_SOURCE_DIR}/tt_adapter
-  COMMAND pip install ${CMAKE_CURRENT_SOURCE_DIR}/model-explorer/src/model-explorer/src/server/package
+  COMMAND pip uninstall -y ai-edge-model-explorer
+  COMMAND ${CMAKE_COMMAND} -E remove_directory ${CMAKE_CURRENT_SOURCE_DIR}/model-explorer/src/model-explorer/src/server/package/build
+  COMMAND pip install --upgrade ${CMAKE_CURRENT_SOURCE_DIR}/model-explorer/src/model-explorer/src/server/package
```
Comment (odjuricicTT) [inline tools/explorer/CMakeLists.txt:23]: Just for my context, why is this needed?

File: tools/explorer/CMakeLists.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@ include(ExternalProject)
 set(TT_EXPLORER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/run.py)
 set(TTMLIR_BUILD_BIN_DIR ${TTMLIR_BINARY_DIR}/bin)
 
-set(MODEL_EXPLORER_VERSION "d0b53c3b7049fd41ea1caff193706272c399fac9")
+set(MODEL_EXPLORER_VERSION "50d3a757c990d30c69d0f9f544b66b0e6af89f81")
```
Comment (odjuricicTT) [inline tools/explorer/CMakeLists.txt:6]: I can't seem to build with this commit:
```
Cloning into 'model-explorer'...
remote: Enumerating objects: 4059, done.
remote: Counting objects: 100% (836/836), done.
remote: Compressing objects: 100% (349/349), done.
remote: Total 4059 (delta 645), reused 487 (delta 487), pack-reused 3223 (from 2)
Receiving objects: 100% (4059/4059), 10.03 MiB | 2.97 MiB/s, done.
Resolving deltas: 100% (2910/2910), done.
fatal: reference is not a tree: 50d3a757c990d30c69d0f9f544b66b0e6af89f81
CMake Error at /localdev/odjuricic/repos/tt-mlir/tools/explorer/model-explorer/tmp/model-explorer-gitclone.cmake:61 (message):
  Failed to checkout tag: '50d3a757c990d30c69d0f9f544b66b0e6af89f81'
```

Probably due to some rebase / squash / merge.

Should we just use tip of main now?

**Review by odjuricicTT — APPROVED** [review]
Awesome! Thanks for this fix that i was overdue in making :)

File: tools/explorer/CMakeLists.txt (hunk 1 above)
Comment (vprajapati-tt) [reply tools/explorer/CMakeLists.txt:23]: Explorer creates some weird build artifacts and I was getting behaviour where cached versions of the frontend would be propagated from build, so this is to clear and delete any build artifacts to prevent that.

File: tools/explorer/CMakeLists.txt (hunk 2 above)
Comment (vprajapati-tt) [reply tools/explorer/CMakeLists.txt:6]: Yeah let me change this to tip.
